apps/navstar_dns/src/navstar_dns_poll_fsm.erl

```diff
@@ -401,17 +400,52 @@ mesos_master_poll() ->
     end.
 
 
+%%
+%% Tasks IPs are identified with up to three DNS A records: agentip, containerip and autoip.
+%% The autoip is derived from the containerip or agentip.  The autoip is defined as matching
+%% the containerip, if it is reachable.  A containerip is reachable if port mappings don't exist
+%% and the containerip exists.
+%%
+%% Pseudo Code
+%%
+%% if If task.network_info.ip_address AND NOT task.port_mappings
+%%     return task.networkinfo.ip_address
+%% else
+%%     return agent_ip
+%%
+
 -spec(task_ip_by_agent(task()) -> {binary(), inet:ip4_address()}).
 task_ip_by_agent(_Task = #task{slave = #slave{pid = #libprocess_pid{ip = IP}}}) ->
     {<<"agentip">>, IP}.
 
+-spec(task_ip_by_agent(task(), binary()) -> {binary(), inet:ip4_address()}).
+task_ip_by_agent(_Task = #task{slave = #slave{pid = #libprocess_pid{ip = IP}}}, Label) ->
+    {Label, IP}.
+
 -spec(task_ip_by_network_infos(task()) -> {binary(), inet:ip4_address()}).
 task_ip_by_network_infos(
     #task{statuses = [#task_status{container_status = #container_status{network_infos = NetworkInfos}}|_]}) ->
     [#network_info{ip_addresses = IPAddresses}|_] = NetworkInfos,
     [#ip_address{ip_address = IP}|_] = IPAddresses,
     {<<"containerip">>, IP}.
 
+-spec(task_ip_by_network_infos(task(), binary()) -> {binary(), inet:ip4_address()}).
+task_ip_by_network_infos(
+    #task{statuses = [#task_status{container_status = #container_status{network_infos = NetworkInfos}}|_]}, Label) ->
+    [#network_info{ip_addresses = IPAddresses}|_] = NetworkInfos,
+    [#ip_address{ip_address = IP}|_] = IPAddresses,
+    {Label, IP};
+task_ip_by_network_infos(Task, Label) ->
+   task_ip_by_agent(Task, Label).
+
+-spec(task_ip_autoip(task()) -> {binary(), inet:ip4_address()}).
+task_ip_autoip(Task = #task{container = #container{type = docker, docker = #docker{port_mappings = []}}}) ->
```

Comment:
Do you want me to wait to merge before that? And you can submit a follow on patch, or good to merge now?